implementation of Gaussian Elimination pivoting as a numerical linear algebra algorithm

TheKidPadra opened the pull request:
### Describe your change:

This is my first ever contribution to Hacktoberfest 2023. I hope I did everything right.
I'm kind of a newbie to Hacktoberfest.
And If I did anything wrong, please tell me, and I'll fix it right away.
I would really appreciate that
Thank you for this opportunity


* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

TheKidPadra commented on linear_algebra/src/gaussianeliminationpivoting.py:
```diff
@@ -0,0 +1,75 @@
+import numpy as np
+from typing import List
+
+def custom_pivoting(a: np.ndarray, n: int, i: int) -> int:
```
I already provided a descriptive name for every parameter

TheKidPadra commented:
Can someone help me I did everything and it says to use list instead of List and it's not possible 
the second thing is I already provided a descriptive name for every parameter but it still says no

cclauss commented:
Click the `Details` links to the right of the failing tests ❌ below for more info.

TheKidPadra commented:
> Click the `Details` links to the right of the failing tests ❌ below for more info.

Thank you, dear [cclauss](https://github.com/cclauss),
Finally, all checks have passed
Waiting for a Review

TheKidPadra commented:
Only 5 days left to end of October can someone please review this code?

cclauss commented:
> Please provide a descriptive name for the parameter: `a`

As discussed in `CONTRIBUTING.md` we want to avoid single-letter variable names.  Please use self-documenting variable names to help your reader understand the algorithm.

TheKidPadra commented:
I changed 'a' to 'matrix' and 'coeff_matrix' for better clarity

cclauss commented on linear_algebra/src/gaussian_elimination_pivoting.py:
```diff
@@ -1,9 +1,9 @@
 import numpy as np
 
-
```
Please put this blank line back.

TheKidPadra replied:
yes sir

TheKidPadra commented:
All things are fixed, please review it before the end of Hacktoberfest.
I changed 'a' to matrix and 
I changed 'n' to num_rows
I changed 'i' to column_index
for better clarity

TheKidPadra commented:
It says FileNotFoundError: matrix.txt not found.

TheKidPadra commented:
It says FileNotFoundError: matrix.txt not found 
but matrix.txt is in the same folder as the source code 
can you help me?

cclauss commented on linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py:
```diff
@@ -0,0 +1,93 @@
+import sys
+import time
```
```suggestion
import time
from pathlib import Path
```

cclauss commented on linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py:
```diff
@@ -0,0 +1,93 @@
+
+import numpy as np
+
+matrixab = np.loadtxt("matrix.txt")
```
```suggestion
matrixab = np.loadtxt(Path(__file__).parent / "matrix.txt")
```

cclauss commented on linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py:
```diff
@@ -0,0 +1,93 @@
+B = np.copy(matrixab[:, matrixab.shape[1] - 1])
+
+
+def foo(matrix):
```
Please.  Come up with a better name for this function and add a type hint and maybe even some doctests.

According to `CONTRIBUTING.md` algorithmic functions should not `print()`.

TheKidPadra replied:
Function is renamed to solve_linear_system, and type hints are added for parameters and return values. The doctest provides an example of usage and asserts the correctness of the solution. I've also removed the print statements and replaced the exit call with a sys.exit exception to allow for better error handling.

cclauss commented on linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py:
```diff
@@ -0,0 +1,93 @@
+    return np.asarray(xlst)
+
+
+vectorofxalpha = foo(matrixab)
```
```suggestion
if __name__ == "__main__":
    vectorofxalpha = foo(matrixab)
```
And indent all lines below in this file.

TheKidPadra commented:
I did everything but it still doesn't pass build test 
but It's working in VScode

TheKidPadra commented:
Will you accept it?

TheKidPadra commented:
Your right sir. I don't know what was I thinking.
Here, finally it got fixed
